branch models-repo-services to dev

dzmitry192 opened the pull request:
were added models, repositories and services(without implemented logic)

dmitriyyy192 committed:
were added models, repositories and services(without implemented logic)

AntonBahd commented on src/main/java/com/innowise/clientservice/model/entity/ClientEntity.java:
```diff
@@ -0,0 +1,36 @@
+import java.util.Set;
+
+@Entity
+@Table(name = "clients", schema = "app")
```
если еще не добавлял индекс на свои таблицы до этого, то как раз можешь сделать это тут, чтобы иметь представление о том, как оно делается. Мб колонка имейла как раз подойдет

AntonBahd commented on src/main/java/com/innowise/clientservice/model/entity/ClientEntity.java:
```diff
@@ -0,0 +1,36 @@
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @NotNull
+    private String firstName;
```
общая рекомендация, советую оставлять над каждым полем энтити аннотацию column с указанием точного имени переменной. Так ты будешь всегда знать, что название у нее одно и то же, даже если в коде поменяется название переменной

AntonBahd commented:
пока всё норм, но тут конечно надо еще логику реализовать, секьюрити и тд. В общем пингуй, когда у тебя сервис полностью и до конца готов

dmitriyyy192 committed:
were added:
- logic for creating user in keycloak
- config for connecting to database
- some changes in models
etc.

dmitriyyy192 committed:
was configured security oauth2.0

dmitriyyy192 committed:
was added:
- crud operations without delete
- some changes in models
- mapper for client entity
- custom exceptions
etc.

dmitriyyy192 committed:
was added:
- client controller
- controller advice for handling messages
- necessary validation

dmitriyyy192 committed:
was added:
- swagger openapi3
- some routes to whitelist for SecurityConfig
- errors dto
- some changes to keycloak service and config
- OpenAPI3Configuration class
etc.

dmitriyyy192 committed:
some changes

dzmitry-sivachenka committed:
were added feign client, changed some logic in client service impl etc.
